src/App.tsx

```diff
@@ -1,14 +1,71 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [isLoaded, setIsLoaded] = useState<boolean>(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [titleFilter, setTitleFilter] = useState<string>('');
+  const [filter, setFilter] = useState<string>('');
+
+  const handleSelectedTodo = (todo: Todo) => {
+    const findTodo = todos.find(value => value === todo) as Todo;
+
+    setSelectedTodo(findTodo);
+  };
+
+  const handleModalClose = () => {
+    setSelectedTodo(null);
+  };
+
+  const handleFilter = (status: string) => {
+    if (status === 'all') {
+      setFilter('all');
+    } else if (status === 'completed') {
+      setFilter('completed');
+    } else if (status === 'active') {
+      setFilter('active');
+    }
+  };
```

Comment:
this can be abstracted outside of the component - please use switch with finite number of cases and make sure you can only pick 3 types of filters